src/PowerShellEditorServices/Server/PsesDebugServer.cs

```diff
@@ -64,17 +68,17 @@ public async Task StartAsync()
                 options.Serializer = new DapProtocolSerializer();
                 options.Reciever = new DapReciever();
                 options.LoggerFactory = _loggerFactory;
-                Extensions.Logging.ILogger logger = options.LoggerFactory.CreateLogger("DebugOptionsStartup");
+                ILogger logger = options.LoggerFactory.CreateLogger("DebugOptionsStartup");
 
                 // We need to let the PowerShell Context Service know that we are in a debug session
                 // so that it doesn't send the powerShell/startDebugger message.
                 _powerShellContextService = ServiceProvider.GetService<PowerShellContextService>();
                 _powerShellContextService.IsDebugServerActive = true;
 
                 // Needed to make sure PSReadLine's static properties are initialized in the pipeline thread.
-                if (!s_hasRunPsrlStaticCtor && _usePSReadLine)
+                if (_usePSReadLine && Interlocked.Exchange(ref s_hasRunPsrlStaticCtor, 1) == 0)
                 {
-                    s_hasRunPsrlStaticCtor = true;
+                    // This must be run synchronously to ensure debugging works
                     _powerShellContextService
                         .ExecuteScriptStringAsync("[System.Runtime.CompilerServices.RuntimeHelpers]::RunClassConstructor([Microsoft.PowerShell.PSConsoleReadLine].TypeHandle)")
                         .Wait();
```

Comment:
Out of scope I know, optional fix here:

```suggestion
                        .GetAwaiter()
                        .GetResult();
```